src/core/annotation.js

```diff
@@ -752,6 +754,26 @@ var TextWidgetAnnotation = (function TextWidgetAnnotationClosure() {
   return TextWidgetAnnotation;
 })();
 
+var ButtonWidgetAnnotation = (function ButtonWidgetAnnotationClosure() {
+  function ButtonWidgetAnnotation(params) {
+    WidgetAnnotation.call(this, params);
+
+    this.data.noToggleToOff = this.hasFieldFlag(
+      AnnotationFieldFlag.NOTOGGLETOOFF);
+    this.data.radio = this.hasFieldFlag(AnnotationFieldFlag.RADIO);
```

Comment:
The main validation should happen in the `core` layer, so wouldn't e.g. this be enough:
```javascript
this.data.radio = this.hasFieldFlag(AnnotationFieldFlag.RADIO) &&
                  !this.hasFieldFlag(AnnotationFieldFlag.PUSHBUTTON);
```